llvm/tools/llvm-lsp/OptRunner.h

```diff
@@ -121,50 +164,80 @@ class OptRunner {
     return FinalIR;
   }
 
+  llvm::Expected<std::pair<SmallString<32>, SmallString<32>>>
+  runShellOpt(std::vector<std::string> Args) {
```

Comment:
Maybe we can take optional arguments for Stderr and Stdout files. This way the caller can ensure it has the output of running opt where it wants without the hassle of intermediate temporary files.